userns.c

```diff
@@ -0,0 +1,371 @@
+/* Copyright © 2020 Arista Networks, Inc. All rights reserved.
+ *
+ * Use of this source code is governed by the MIT license that can be found
+ * in the LICENSE file.
+ */
+
+#include <err.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "userns.h"
+
+enum {
+	ID_MAX     = 65534,
+
+	/* uid/gid values and user/group names are within 32 digits (man useradd) */
```

Comment:
nit: comment seems a bit off - You mean 32 characaters for names, right? I'm not sure if the uid/gid thing applies here at all - the only reference to 32 I can find in useradd(8) is the limit on name lengths

Reply:
Right, I means that this is both an upper bound for usernames and stringified IDs, because linux has 32-bit [ug]ids, which at most has 10 digits in base 10.